### PR TITLE
Use shared GitHub Action for Ruby tests

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -6,82 +6,16 @@ on:
...
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake db:create
-          bundle exec rake db:migrate
-          bundle exec rake db:test:prepare
-      - name: Run plugin tests
-        if: github.event_name != 'push'
-        run: |
-          bundle exec rake test:foreman_leapp -v --trace
-          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb" -v --trace
```
Comment (evgeni) [inline .github/workflows/ruby.yml:80]: https://github.com/theforeman/actions/pull/16

File: lib/foreman_leapp/engine.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
 
+require 'foreman_ansible'
+require 'foreman_remote_execution'
```
Comment (evgeni) [inline lib/foreman_leapp/engine.rb:4]: this fixes https://github.com/theforeman/foreman_leapp/issues/120

(technically, I guess, `foreman_ansible` would be enough, but I felt like being explicit here)

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -6,82 +6,16 @@ on:
...
-        with:
-          ruby-version: 2.7
-          bundler-cache: true
-      - name: Run rubocop
-        if: github.event_name != 'push'
-        run: bundle exec rubocop --format github
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+    with:
+      command: bundle exec rubocop --format github
```
Comment (ekohl) [inline .github/workflows/ruby.yml:14]: While we're changing things:
```suggestion
      command: bundle exec rubocop --parallel --format github
```

**Review by ekohl — COMMENTED** [review]
I think the require statement is correct. I'm not sure how else it could be solved.

**Comment by stejskalleos** [comment]
Thanks @evgeni @ekohl

**Pull request merged**